### PR TITLE
feat(sdk/python): start oomd asynchronously at initialisation

### PR DESCRIPTION
**Pull request by lianxmfor**
Start oomd asynchronously at initialisation

### CODE REVIEW

**Comment by YikSanChan** [comment]
@lianxmfor can you keep the format unchanged in this PR, and format all Python code in another PR? To make code review easier

**Comment by YikSanChan** [comment]
> @lianxmfor can you keep the format unchanged in this PR, and format all Python code in another PR? To make code review easier

ignore this, it's fine.

File: sdk/python/client.py [hunk 1]
```diff
@@ -1,51 +1,130 @@
...
-# Convert google.protobuf.pyext._message.MessageMapContainer object to Python dictionary
 def map_container_to_dict(map_container):
-    return dict({k: getattr(v, v.WhichOneof('kind')) for k, v in map_container.items()})
+    return dict({k: getattr(v, v.WhichOneof("kind")) for k, v in map_container.items()})
+
 
 class Client(object):
-    def __init__(self):
-        pass
+    def __init__(self, port: str, config_path: str):
```
Comment (YikSanChan) [inline sdk/python/client.py:14]: port: int?

File: sdk/python/client.py [hunk 2]
```diff
@@ -1,51 +1,130 @@
 import grpc
 import codegen.oomd_pb2
 import codegen.oomd_pb2_grpc
+import time
+from subprocess import Popen
+
 
-# Convert google.protobuf.pyext._message.MessageMapContainer object to Python dictionary
```
Comment (YikSanChan) [inline sdk/python/client.py:5]: keep the comments

File: sdk/python/client.py [hunk 3]
```diff
@@ -1,51 +1,130 @@
...
 class Client(object):
-    def __init__(self):
-        pass
+    def __init__(self, port: str, config_path: str):
+        try:
+            self.oomd = Popen(["oomd", config_path])
+        except e:
+            raise e
+
+        # wait oomd started
```
Comment (YikSanChan) [inline sdk/python/client.py:20]: wait for oomd to start

File: sdk/python/client.py [hunk 4]
```diff
@@ -1,51 +1,130 @@
...
+        time.sleep(2)
+        channel = grpc.insecure_channel("localhost:%d" % port)
```
Comment (YikSanChan) [inline sdk/python/client.py:22]: what if the channel is disconnected halfway? Is there a way to recover?

If not, I would prefer leaving the channel and stub initialization to each method call.

File: sdk/python/client.py [hunk 5]
```diff
@@ -1,51 +1,130 @@
...
+        self.stub = codegen.oomd_pb2_grpc.OomDStub(channel)
+
+    def __del__(self):
+        self.oomd.terminate()
```
Comment (YikSanChan) [inline sdk/python/client.py:26]: what does terminate do?

Comment (lianxmfor) [reply sdk/python/client.py:26]: stop oomd process while py has done.

Comment (YikSanChan) [reply sdk/python/client.py:26]: gotcha

File: sdk/python/client.py (hunk 3 above)
Comment (lianxmfor) [reply sdk/python/client.py]: fixed

File: sdk/python/client.py (hunk 2 above)
Comment (lianxmfor) [reply sdk/python/client.py:5]: done

File: sdk/python/client.py (hunk 1 above)
Comment (lianxmfor) [reply sdk/python/client.py]: good catch! Surprisingly, no errors were reported when running...

Comment (lianxmfor) [reply sdk/python/client.py]: fixed

File: sdk/python/client.py (hunk 4 above)
Comment (lianxmfor) [reply sdk/python/client.py]: sound good, done.

File: sdk/python/client.py [hunk 6]
```diff
@@ -1,51 +1,142 @@
...
+    return dict({k: getattr(v, v.WhichOneof("kind")) for k, v in map_container.items()})
+
 
 class Client(object):
-    def __init__(self):
-        pass
+    def __init__(self, port: int, config_path: str):
+        try:
+            self.oomd = Popen(["oomd", "--config", config_path])
+        except e:
```
Comment (YikSanChan) [inline sdk/python/client.py:17]: `try ... except e: raise e`

doesn't seem to do anything 🤔

**Review by YikSanChan — APPROVED** [review]
lgtm

File: sdk/python/client.py (hunk 6 above)
Comment (YikSanChan) [reply sdk/python/client.py:17]: If this is the case, just remove `try .. except ...` and let it fail

Comment (lianxmfor) [reply sdk/python/client.py]: Some changes were made

**Review by YikSanChan — APPROVED** [review]
lgtm

**Pull request merged**